code/datums/components/largeobjecttransparency.dm

```diff
@@ -0,0 +1,107 @@
+///Makes large icons partially see through if high priority atoms are behind them.
+/datum/component/largetransparency
+	//Can be positive or negative. Determines how far away from parent the first registered turf is.
+	var/x_offset
+	var/y_offset
+	//Has to be positive or 0.
+	var/x_size
+	var/y_size
+	//The alpha values this switches in between.
+	var/initial_alpha
+	var/target_alpha
+	//if this is supposed to prevent clicks if it's transparent.
+	var/toggle_click
+	var/list/registered_turfs
+	var/amounthidden = 0
+
+/datum/component/largetransparency/Initialize(_x_offset = 0, _y_offset = 1, _x_size = 0, _y_size = 1, _initial_alpha = null, _target_alpha = 140, _toggle_click = TRUE)
+	if(!isatom(parent))
+		return COMPONENT_INCOMPATIBLE
+	x_offset = _x_offset
+	y_offset = _y_offset
+	x_size = _x_size
+	y_size = _y_size
+	if(isnull(_initial_alpha))
+		var/atom/at = parent
+		initial_alpha = at.alpha
+	else
+		initial_alpha = _initial_alpha
+	target_alpha = _target_alpha
+	toggle_click = _toggle_click
+	registered_turfs = list()
+
+
+/datum/component/largetransparency/Destroy()
+	registered_turfs.Cut()
+	return ..()
+
+/datum/component/largetransparency/RegisterWithParent()
+	RegisterSignal(parent, COMSIG_MOVABLE_MOVED, .proc/OnMove)
+	RegisterWithTurfs()
+
+/datum/component/largetransparency/UnregisterFromParent()
+	UnregisterSignal(parent, COMSIG_MOVABLE_MOVED)
+	UnregisterFromTurfs()
+
+/datum/component/largetransparency/proc/RegisterWithTurfs()
+	var/turf/current_turf = get_turf(parent)
+	if(!current_turf)
+		return
+	var/turf/lowleft_tu = locate(clamp(current_turf.x + x_offset, 0, world.maxx), clamp(current_turf.y + y_offset, 0, world.maxy), current_turf.z)
+	var/turf/upright_tu = locate(min(lowleft_tu.x + x_size, world.maxx), min(lowleft_tu.y + y_size, world.maxy), current_turf.z)
+	registered_turfs = block(lowleft_tu, upright_tu) //small problems with z level edges but nothing gamebreaking.
+	//register the signals
+	for(var/registered_turf in registered_turfs)
+		if(!registered_turf)
+			continue
```

Comment:
can this every be true? I dont see you removing items from the list during the loop?